docs/contributors/release.md

```diff
@@ -186,25 +186,55 @@ If you don't have access to [make.wordpress.org/core](https://make.wordpress.org
+The Gutenberg repository mirrors the WordPress SVN repository in terms of branching for each SVN branch, a corresponding Gutenberg `wp/*` branch is created:
 
-### Major WordPress Releases
+ - The `wp/trunk` branch contains all the packages that are published and used in the `trunk` branch of WordPress.
+ - A Gutenberg branch targetting a specific WordPress release and its minors is created (example `wp/5.2`) based on the `wp/trunk` Gutenberg branch when the WordPress `trunk` branch is marked as "feature-freezed". (This usually happens when the first `beta` of the next WordPress major version is released).
 
-For major WordPress releases, the last Gutenberg plugin release is merged into `g-minor`. This involves the following steps:
+### Synchronizing WordPress Trunk
+
+For each Gutenberg plugin release, WordPress trunk should be synchronized with this release. This involves the following steps:
+
+**Note:** The WordPress `trunk` branch can be closed or in "feature-freeze" mode. Usually, this happens between the first `beta` and the first `RC` of the WordPress release cycle. During this period, the Gutenberg plugin releases should not be synchronized with WordPress Core.
+
+1. Ensure the WordPress `trunk` branch is open for enhancements. 
+2. Check out the last published Gutenberg release branch `git checkout release/x.x`
+3. Create a Pull Request from this branch targetting `wp/trunk`.
+4. Merge the Pull Request.
+
+Now, the branch is ready to be used to publish the npm packages.
+
+1. Check out the `wp/trunk` branch.
+2. Run the [package release process] but when asked for the version numbers to choose for each package, (assuming the package versions are written using this format `major.minor.patch`) make sure to bump at least the `minor` version number. For example, if the CHANGELOG of the package to be released indicates that the next unreleased version is `5.6.1`, choose `5.7.0` as a version.
+3. Update the `CHANGELOG.md` files of the published packages with the new released versions and commit to the `wp/trunk` branch.
+4. Cherry-pick the "Publish" (created by Lerna) and the CHANGELOG update commits into the `master` branch of Gutenberg.
+
+Now, the npm packages should be ready and a patch can be created and commited into WordPress `trunk`.
 
-1. Check out the last published Gutenberg release branch `git checkout release/x.x`
-2. Create a Pull Request from this branch into `g-minor`.
-3. Merge the branch.
 
 ### Minor WordPress Releases
 
-For minor releases, the critical fixes targeted for this WordPress Minor release should be cherry-picked into the `g-minor` branch one by one in their chronological order.
+The following workflow is needed when bug fixes or security releases need to be backported into WordPress Core. This can happen in a few use-cases:
```

Comment:
FYI: To be aware of, WordPress currently backports when possible all the way back to WP 3.7

This raises the possibility that Gutenberg may also require patching back to `5.0`

For example, if a security issue is discovered later today and is proposed to be fixed in `5.1.1` it is conceivably possible that the _packages_ be updated will have to be backported to WordPress versions `5.0` and `5.1` as such the Gutenberg branches `wp/5.0`, `wp/5.1`, and `wp/trunk` will **all** require updating at this time.

Reply:
That's exactly why we need to leave "room" for bug fixes and security fixes in the package version number.

Reply:
Package version numbers cannot work like that as semantic versioning dictates that once `6.7.0` is published, `6.6.9` cannot be published, nothing less than `6.7.0` can be published.

Reply:
> semantic versioning dictates that once `6.7.0` is published, `6.6.9` cannot be published,

What makes you say that? How do npm package maintainers publish security fixes for old major and minor versions without forcing people to upgrade?

Reply:
They don't, people are forced to upgrade to the version with the security fix.

For example, if a vulnerability is discovered in version `1.1.1` of a package, and that same package is now at version `7.8.9`, the fix for the `1.1.1` vulnerability must be released in either `7.8.10`, `7.9.0`, or `8.0.0`.

Reply:
I think you're wrong, here's an example https://reactjs.org/blog/2018/08/01/react-v-16-4-2.html

Reply:
This is where trying to align semantic versioning of packages with the WP release versioning clashes.  The problem here is that a latest release of a package could result in new features to the WP version the security fix is for.  This could be mitigated some by the use of the new feature flags functionality in GB but does complicate things.

Reply:
> Package version numbers cannot work like that as semantic versioning dictates that once `6.7.0` is published, `6.6.9` cannot be published, nothing less than `6.7.0` can be published.

I just read through the [semver specs](https://semver.org/) and I'm not seeing this as something that is codified.  Is this something that is documented as a rule somewhere?

Reply:
> Package version numbers cannot work like that as semantic versioning dictates that once `6.7.0` is published, `6.6.9` cannot be published, nothing less than `6.7.0` can be published.

I don't think the spec has an opinion on this, but I've never read it that way. If so, something like PHP isn't following semver. Maintaining LTS for a specific minor release wouldn't be possible. 

I always interpreted the spec as any individual version is just a comparison of the versions that are lower than it, regardless of release date.

Reply:
@nerrad @earnjam I was not inferring a SemVer spec, I was inferring it as a restriction of the npm platform, that said:

@youknowriad I. Did. Not. Know. That. 💥

(I've just confirmed with npm that this can be done, also advised not to publish the _backported_ releases with the `latest` npm tag.

That will make things far simpler, though thinking about the "room" that we would leave behind, if version `1.1.1` of a package is already published, we bump the _minor_ so that it becomes version `1.2.0`, if we then have to patch version `1.1.1` for a vulnerability, by publishing the fix in `1.1.2` is that ok for semantic versioning? There is the possibility the "fix" breaks _backward compatibility which means the package should have been bumped by the _minor_ version and not the _patch_ version 🤔